### PR TITLE
Add Rollouts models

### PR DESCRIPTION
**Pull request by Mrod1598**
### Proposed Change
- Add rollouts models

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: model/resource_test.go [hunk 1]
```diff
@@ -23,8 +25,54 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	// testConfigCommonConfig = config.Config{
```
Comment (dsvanlani) [inline model/resource_test.go:32]: Is this still neccessary?

**Review by dsvanlani — APPROVED** [review]
Unsure about that comment - was this in the Rollouts upstream branch?

File: model/configuration.go [hunk 2]
```diff
@@ -106,12 +115,232 @@ type ConfigurationSpec struct {
...
+
+func (s RolloutStatus) String() string {
+	str, ok := rolloutStatusMap[s]
+	if !ok {
+		return "none"
+	}
+	return str
+}
+
+// TODO make these strings nice
```
Comment (cpheps) [inline model/configuration.go:195]: We should ask if this TODO is still valid or if we're happy with these

**Review by cpheps — APPROVED** [review]

File: model/configuration.go (hunk 2 above)
Comment (Mrod1598) [reply model/configuration.go:195]: tossed a comment on the original pr

**Pull request merged**